src/controllers/cartController.js

```diff
@@ -0,0 +1,180 @@
+  } catch (error) {
+    next(error);
+  }
+};
+
+// add book to catr
+const addToCart = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+    const { bookId } = req.body;
+
+    const book = await Book.findById(bookId);
+    if (!book || !book.isAvailable) {
+      throw new BadRequestError('Book is unavailable or does not exist');
+    }
+
+    let cart = await Cart.findOne({ createdBy: userId });
+    if (!cart) {
+      cart = await Cart.create({ createdBy: userId });
+    }
+
+    const exists = cart.orderItems.some(
+      (item) => item.book.toString() === bookId
+    );
+    if (exists) throw new BadRequestError('Book is already in the cart');
+
+    cart.orderItems.push({
+      book: book._id,
+      title: book.title,
+      author: book.author,
+      coverImageUrl: book.coverImageUrl,
+      price: book.price,
+      isAvailable: book.isAvailable,
+    });
+
+    
+    cart.total = calculateCartTotal(cart);
+
+    await cart.save();
+    res.status(201).json({ message: 'Book added to cart', cart });
+  } catch (error) {
+    next(error);
+  }
+};
+
+
+// delete from cart
+const deleteFromCart = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+    const { id: cartItemId } = req.params;
+
+    const cart = await Cart.findOne({ createdBy: userId });
+    if (!cart) {
+      throw new NotFoundError('Cart not found');
+    }
+
+    const filteredItems = cart.orderItems.filter(
+      (item) => item._id.toString() !== cartItemId
+    );
+    if (filteredItems.length === cart.orderItems.length) {
+      throw new NotFoundError('Book not found in the cart');
+    }
+
+    cart.orderItems = filteredItems;
+
+    cart.total = calculateCartTotal(cart);
+
+    await cart.save();
+    res
+      .status(StatusCodes.OK)
+      .json({ message: 'Item removed successfully', cart });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const createPaymentIntent = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+    const cart = await Cart.findOne({ createdBy: userId }).populate(
+      'orderItems.book'
+    );
+
+    if (!cart || cart.orderItems.length === 0) {
+      throw new BadRequestError('Your cart is empty');
+    }
+
+    if (cart.clientSecret) {
+      return res
+        .status(StatusCodes.OK)
+        .json({ clientSecret: cart.clientSecret });
+    }
+
+    const totalAmount = Math.round(cart.total * 100);
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount: totalAmount,
+      currency: 'usd',
+    });
+
+    cart.clientSecret = paymentIntent.client_secret;
+    cart.paymentIntentId = paymentIntent.id;
+    await cart.save();
+
+    res
+      .status(StatusCodes.OK)
+      .json({ clientSecret: paymentIntent.client_secret });
+  } catch (error) {
+    next(error);
+  }
+};
+
+// Confirm payment
+const confirmPayment = async (req, res, next) => {
+  try {
+    const { paymentIntentId } = req.body;
+
```

Comment:
Do we handle partial payments or retries on payment failure? 

Can add that later if wanted.

Reply:
Yes thank you,  will finish payment implementation in next ticket